lib/chore/queues/sqs/consumer.rb

```diff
@@ -114,11 +114,12 @@ def queue_timeout
 
         # Access to the configured SQS connection object
         def sqs
-          @sqs ||= AWS::SQS.new(
+          sqs_options = {
             :access_key_id => Chore.config.aws_access_key,
-            :secret_access_key => Chore.config.aws_secret_key,
-            :logger => Chore.logger,
-            :log_level => :debug)
+            :secret_access_key => Chore.config.aws_secret_key
+          }
+          sqs_options.merge!(:logger => Chore.logger, :log_level => :debug) if Chore.config.log_level == Logger::DEBUG
```

Comment:
If it worked, yes. Near as I can tell despite having a `log_level` setting,
the AWS gem (or at least the SQS portion) either ignores it entirely, or
uses at least `info` level logging for everything. There is no difference
between `info` and `debug` in my testing.

On Tue, Dec 8, 2015 at 3:56 PM, StabbyCutyou <notifications@github.com>
wrote:

> In lib/chore/queues/sqs/consumer.rb
> <https://github.com/Tapjoy/chore/pull/11#discussion_r47012287>:
>
> >              :access_key_id => Chore.config.aws_access_key,
> > -            :secret_access_key => Chore.config.aws_secret_key,
> > -            :logger => Chore.logger,
> > -            :log_level => :debug)
> > +            :secret_access_key => Chore.config.aws_secret_key
> > +          }
> > +          sqs_options.merge!(:logger => Chore.logger, :log_level => :debug) if Chore.config.log_level == Logger::DEBUG
>
> Would it not be best to keep having it use the Chore.logger, and simply
> default it's internal log level to match what chore is configured to do?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Tapjoy/chore/pull/11/files#r47012287>.
>